language/move-stdlib/sources/Option.move

```diff
@@ -178,6 +178,22 @@ module Std::Option {
         ensures borrow(t) == e;
     }
 
+    /// Swap the old value inside `t` with `e` and return the old value;
+    /// or if there is no old value, fill it with `e`.
+    /// Different from swap(), swap_or_fill() allows for `t` not holding a value.
+    public fun swap_or_fill<Element>(t: &mut Option<Element>, e: Element): Option<Element> {
```

Comment:
fwiw, Rust calls this `replace`. Which seems like a pretty good name to me. 

Thoughts @tzakian?

Reply:
But the swap() in move-stdli is different from rust's swap(). So if we rename this to replace, it might make swap confusing.